flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/sink/HBaseSinkFunction.java

```diff
@@ -191,8 +195,10 @@ private void checkErrorAndRethrow() {
     public void invoke(T value, Context context) throws Exception {
         checkErrorAndRethrow();
 
-        mutator.mutate(mutationConverter.convertToMutation(value));
-
+        Mutation mutation = mutationConverter.convertToMutation(value);
+        synchronized (mutationMap) {
+            mutationMap.put(mutation.getRow(), mutation);
```

Comment:
One potential bug might arise when the `mutation.getRow()` returns an array. As we know, the `hashCode` and `equals` of two different array instances are different regardless of whether their contents are identical.

Reply:
To overcome this, I suggest to convert it to a base64 string, e.g.:
```java
String key = Base64.getEncoder().encodeToString(mutation.getRow());
```

Or to create a simple wrapper class, where the `equals` and `hashCode` are overridden properly for arrays.

Reply:
thank you for yours suggestion, I will improve it.